### PR TITLE
Add the module path to the definition name to avoid name collisions

### LINKED ISSUE #357: [Actix] Type definition overriden when struct used with same name multiple times
Hey there,

Thanks for all the hard work on Paperclip. I'm using it together with Actix to generate a spec for my API, however I've noticed something odd. Almost every request handler (split out into modules) has a `Request` and `Response` struct, for the request and response respectively. However it seems to be only provided once in the final spec. 

Assume I have the following two endpoints:
```rs
// foo.rs
#[derive(Deserialize, Apiv2Schema)]
pub struct Request {
    a: i32,
    b: i32
}
...

// bar.rs
#[derive(Deserialize, Apiv2Schema)]
pub struct Request {
    c: String,
    d: f64
}
```
Both API will show the same request struct, rather than foo having it's own, and bar having it's own

## Potential solution
I think a potential solution would be to include the module in the definition name, that way you can never have overlap

## Versions
actix: 3.3.2
paperclip: 0.6.1
rustc: 1.57.0-beta

### PR DESCRIPTION
**Pull request by TobiasDeBruijn**
This PR intends to fix #357, but uses an alternative approach to #358.

With this PR, the behaviour will be as follows:
```rs
mod foo {
    pub mod bar {
        use paperclip::actix::{web, api_v2_operation, Apiv2Schema};
        use serde::Serialize;

        #[derive(Serialize, Apiv2Schema)]
        pub struct Response {
            bar: i32
        }

        #[api_v2_operation]
        pub fn b() -> web::Json<Response> {
            web::Json(Response { bar: 10 })
        }
    }
}
```
results in the following spec:
```yaml
  foo_bar_Response:
    type: object
    properties:
      bar:
        type: integer
        format: int32
    required:
      - bar
...
          schema:
            $ref: '#/definitions/foo_bar_Response' 
```
The advantage of this PR over #358 is that the generated spec is always the same for the same input code.

The name is determined by taking the output of `std::module_path!()`, removing the first 'component' (i.e. the crate name), then joining with `_` as a separator.

This **will not** break code backwards compatibility, but it **will** break spec backwards compatibility, i.e. the generated spec is different from how it would be without this PR

### CODE REVIEW

File: macros/src/actix.rs [hunk 1]
```diff
@@ -720,26 +721,42 @@ pub fn emit_v2_definition(input: TokenStream) -> TokenStream {
...
+                    .filter(|(index, _)| *index != 0) // Skip the first element, i.e the crate name
+                    .map(|(_, component)| component)
+                    .collect::<Vec<_>>()
+                    .join("_");
+                format!("{}_{}", trimmed_module_path, #schema_name)
+            }
+        }
 
+        impl #impl_generics paperclip::v2::schema::Apiv2Schema for #name #ty_generics #where_clause {
+            //const NAME: Option<&'static str> = None;
```
Comment (tiagolobocastro) [inline macros/src/actix.rs:742]: can remove this line?

**Review by tiagolobocastro — COMMENTED** [review]
Hey @TobiasDeBruijn, as discussed previously this should be opt-in only - we can probably do that by hiding this behind a feature gate?

**Comment by tiagolobocastro** [comment]
Looks good, can we add a test for this feature?

**Comment by TobiasDeBruijn** [comment]
I'll give writing a test a shot!

**Pull request merged**